digitalidentity/service.go

```diff
@@ -20,26 +21,61 @@ type SessionResult struct {
 }
 
 // CreateShareSession creates session using the supplied session specification
-func CreateShareSession(httpClient requests.HttpClient, shareSession *ShareSession, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ShareURL, err error) {
+func CreateShareSession(httpClient requests.HttpClient, shareSession *ShareSession, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share SessionResult, err error) {
 	endpoint := identitySesssionCreationEndpoint
 
 	payload, err := shareSession.MarshalJSON()
 	if err != nil {
 		return share, err
 	}
 
+	headers := requests.AuthHeader(clientSdkId)
+
 	request, err := requests.SignedRequest{
 		Key:        key,
 		HTTPMethod: http.MethodPost,
 		BaseURL:    apiUrl,
 		Endpoint:   endpoint,
-		Headers:    nil,
+		Headers:    headers,
 		Body:       payload,
 	}.Request()
 	if err != nil {
 		return share, err
 	}
 
+	response, err := requests.Execute(httpClient, request, ShareURLHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		//fmt.Printf("err 2:=> %s\n\r", err)
+		return share, err
+	}
+
+	defer response.Body.Close()
+
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return share, err
+	}
+
+	err = json.Unmarshal(responseBytes, &share)
+
+	return share, err
+}
+
+// GetSession get session info using the supplied sessionID
+func GetSession(httpClient requests.HttpClient, sessionID string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ShareSessionResult, err error) {
+	endpoint := identitySesssionCreationEndpoint
+	headers := requests.AuthHeader(clientSdkId)
+	request, err := requests.SignedRequest{
+		Key:        key,
+		HTTPMethod: http.MethodGet,
+		BaseURL:    apiUrl,
+		Endpoint:   endpoint,
+		Headers:    headers,
+	}.Request()
+	if err != nil {
+		return share, err
+	}
+
```

Comment:
can't comment on the line below, but ShareURLHTTPErrorMessages is not the correct mapping